### ISSUE TITLE
Creating JPG thumbnails from PDF causes problems with version 6.8.9 of ImageMagick

### ISSUE THREAD

**Issue by bharatmaher** [ImageMagick/ImageMagick#1508]
### Description
I am using Wordpress 5.1 and ImageMagick 6.8.9. When I upload PDF from the Media menu of WordPress then it creates JPG thumbnail with a black background.

 First, I used default media import functionality of WP to import PDF files then I tried with my own script to create JPG thumbnail of the first page. Here it is  

`$uploads = wp_upload_dir();`
`$imagick = new Imagick();`
`$imagick->readImage('test.pdf[0]');`
` $imagick->trimImage(2);`
`$imagick->setImagePage(0, 0, 0, 0);`
`$imagick->setImageFormat('jpeg');`
`file_put_contents($uploads['path'] . '/thumbnail.png', $imagick);`

In both cases, it creates an image with a black background.
Does anyone have any idea about this?

**Comment by fmw42** [comment]
JPEG does not support transparency. Change your output to PNG or TIFF

**Comment by urban-warrior** [comment]
Try setting the background color to white before you write it, perhaps $imagick->setBackground('white');

**Comment by bharatmaher** [comment]
> Try setting the background color to white before you write it, perhaps $imagick->setBackground('white');

@urban-warrior ,
 I tried to set background color but no luck. I tried two options.
`$imagick->setBackground('white');`
And
`$imagick->setBackgroundColor(new ImagickPixel('transparent'));`

**Comment by dlemstra** [comment]
Did you set it before you are reading the image?

**Comment by bharatmaher** [comment]
> Did you set it before you are reading the image?

@dlemstra ,
Yes. Actually, I tried both before and after reading just to make sure if it works.
I observe this issue only when I have transparent PDF background.

**Comment by fmw42** [comment]
If your PDF is CMYK with transparency, you must convert to RGB before reading the PDF using -colorspace sRGB. Ghostscript cannot handle transparent CMYK PDF files.
